### PR TITLE
W-13038823 - Groovy Resource Releaser (#12571)

### PR DESCRIPTION
**Pull request by acurci-at-mulesoft**

### CODE REVIEW

File: modules/artifact/src/main/java/org/mule/module/artifact/classloader/GroovyResourceReleaser.java
Comment (asanguinetti) [inline modules/artifact/src/main/java/org/mule/module/artifact/classloader/GroovyResourceReleaser.java:29]:
```suggestion
 * @since 4.3.0
```
We'll need to change this if we backport it to 4.3.

**Review by asanguinetti — CHANGES_REQUESTED** [review]
Just a small JDoc change.

Additionally, I have the understanding that we are not allowed to merge changes to versions that [are outside standard support](https://www.mulesoft.com/legal/versioning-back-support-policy#support-matrix) without VP approval.

**Comment by facunlk** [comment]
@asanguinetti this change is approved, if the review is ok, we can move forward with it.

**Review by asanguinetti — APPROVED** [review]

**Pull request merged**
